gst/mqtt/mqttsink.c

```diff
@@ -1067,6 +1120,103 @@ gst_mqtt_sink_set_mqtt_qos (GstMqttSink * self, const gint qos)
   self->mqtt_qos = qos;
 }
 
+/**
+ * @brief Getter for the 'ntp-sync' property.
+ */
+static gboolean
+gst_mqtt_sink_get_mqtt_ntp_sync (GstMqttSink * self)
+{
+  return self->mqtt_ntp_sync;
+}
+
+/**
+ * @brief Setter for the 'ntp-sync' property
+ */
+static void
+gst_mqtt_sink_set_mqtt_ntp_sync (GstMqttSink * self, const gboolean flag)
+{
+  self->mqtt_ntp_sync = flag;
+}
+
+/**
+ * @brief Getter for the 'ntp-srvs' property.
+ */
+static gchar *
+gst_mqtt_sink_get_mqtt_ntp_srvs (GstMqttSink * self)
+{
+  return self->mqtt_ntp_srvs;
+}
+
+/**
+ * @brief Setter for the 'ntp-srvs' property
+ */
+static void
+gst_mqtt_sink_set_mqtt_ntp_srvs (GstMqttSink * self, const gchar * pairs)
+{
+  gchar **pair_arrs = NULL;
+  guint hnum = 0;
+  gchar *pair;
+  guint i, j;
+
+  if (g_strcmp0 (self->mqtt_ntp_srvs, pairs) == 0)
+    return;
+
+  g_free (self->mqtt_ntp_srvs);
+  self->mqtt_ntp_srvs = g_strdup (pairs);
+
+  pair_arrs = g_strsplit (pairs, ",", -1);
+  if (pair_arrs == NULL)
+    return;
+
+  hnum = g_strv_length (pair_arrs);
+  if (hnum == 0)
+    goto err_free_pair_arrs;
+
+  g_free (self->mqtt_ntp_hnames);
+  self->mqtt_ntp_hnames =
+      g_try_malloc0 (hnum * sizeof (*self->mqtt_ntp_hnames));
+  if (!self->mqtt_ntp_hnames)
+    goto err_free_pair_arrs;
+
+  g_free (self->mqtt_ntp_ports);
+  self->mqtt_ntp_ports = g_try_malloc0 (hnum * sizeof (self->mqtt_ntp_ports));
+  if (!self->mqtt_ntp_ports)
+    goto err_free_mqtt_ntp_hnames;
+
+  self->mqtt_ntp_num_srvs = hnum;
+  for (i = 0, j = 0; i < hnum; i++) {
+    gchar **hname_port;
+    gchar *hname;
+    gchar *eport;
+    gulong port_ul;
+
+    pair = pair_arrs[i];
+    hname_port = g_strsplit (pair, ":", 2);
+    hname = hname_port[0];
+    port_ul = strtoul (hname_port[1], &eport, 10);
+    if ((port_ul == 0) || (port_ul > UINT16_MAX)) {
+      self->mqtt_ntp_num_srvs--;
+    } else {
+      self->mqtt_ntp_hnames[j++] = g_strdup (hname);
```

Comment:
1200 and 1201 both increases index (j++), this may raise exception.

Reply:
this potentially causes out-of-bounds... and looks like an error.

Reply:
My bad. It is fixed.